python/cudf/cudf/_lib/cpp/hash.pxd

```diff
@@ -44,3 +49,9 @@ cdef extern from "cudf/hashing.hpp" namespace "cudf::hashing" nogil:
         const table_view& input,
         const uint64_t seed
     ) except +
+
+    cpdef enum class hash_id(int):
```

Comment:
We are removing this enum in https://github.com/rapidsai/cudf/pull/15375. It's being moved to the hash partitioning code, but with only IDENTITY and MURMUR3 options. Please check the features in this PR against that one, to make sure we avoid conflicts.